pykern/util.py

```diff
@@ -65,3 +67,59 @@ def _check_dev_files(root):
         # Don't run from an install directory
         r = pkio.py_path()
     return pkio.mkdir_parent(r.join(_DEFAULT_ROOT))
+
+
+def is_pure_text(value, is_truncated=False):
+    """Guesses if value is text data using heuristics:
+
+    Checks if value can be utf-8 decoded.
+
+    If fails to decode and is_truncated, probes backwards up to 4 chars
+    (4 is maximum length of a utf-8 char) in case a valid utf-8
+    char was truncated at the boundary of value.
+
+    Returns False if null byte is present.
+
+    On successful decode, checks that the amount of control codes not
+    typical of text data do not exceed one third of the total characters.
+
+    Args:
+        value (bytes): bytes data
+        is_truncated (bool): whether or not value has been truncated
+
+    Returns:
+        bool: True if bytes_data is likely pure text, false if likely binary
+    """
+
+    def _is_accepted_control_code_ratio(text_value):
+        c = 0
+        for char in text_value:
+            if ord(char) == 0:
+                return False
+            if ord(char) < 32 and ord(char) not in _VALID_ASCII_CONTROL_CODES:
+                c += 1
+        return (c / len(text_value)) < _ACCEPTABLE_CONTROL_CODE_RATIO
+
+    def _try(chunk):
+        try:
+            return chunk.decode("utf-8", "strict")
+        except UnicodeDecodeError:
+            return False
+
+    def _unicode_decoded(value):
```

Comment:
`_unicode_decoded` is `_ut8_decoded`. Technically Unicode is the general standard for utf8, utf16, etc.